content/Hardware Support/Generic/Reset-the-flash-memory-on-STM32H747-based-devices.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "Reset the flash memory on STM32H747-based devices"
+---
+
+Learn how to reset the QSPI flash memory for STM32H747-based Arduino devices by uploading a utility sketch. The sketch will remove any previous partitioning scheme, and recreate the factory-defined partition scheme. Finally, the latest connectivity firmware and TLS certificates are reinstalled.
+
```

Comment:
Maybe we can add more detail like so:

"This procedure is required, for example, when Opta or PMC have been previously configured to work on PLC IDE memory scheme and then the user wishes to "go back" to original setting to connect the device on regular Arduino IDE"

Reply:
@i-herrera More detail about when the procedure is useful would be great :)

> when Opta or PMC have been previously configured to work on PLC IDE

Does this refer to the "download runtime" action in PLC IDE?

> to "go back" to original setting to connect the device on regular Arduino IDE"

As written, the instructions requires that the Opta is detected by Arduino IDE/Cloud Editor so that the sketch can be uploaded. Will this not be possible if the PLC IDE runtime has been downloaded?

Reply:
> Does this refer to the "download runtime" action in PLC IDE?
Yes, as soon as runtime is downloaded the memory partition changes. 

> As written, the instructions requires that the Opta is detected by Arduino IDE/Cloud Editor so that the sketch can be uploaded. Will this not be possible if the PLC IDE runtime has been downloaded?
Yes, it is possible. Even with the runtime uploaded the device will still be detected in Arduino IDE and regular sketch upload is possible. The memory partition sketch is recommended to reset the memory allocation affecting other peripherals such as wifi

Reply:
Resolved by d77a580a99fd76890d1a98d600746532ff634968